src/wfdlgs3.c

```diff
@@ -1699,3 +1714,63 @@ DestroyCancelWindow()
    }
    CancelInfo.hCancelDlg = NULL;
 }
+
+//
+// GetProductVersion
+// Gets the product version values for the current module
+//
+// Parameters:
+//   pwMajor    - [OUT] A pointer to the major version number
+//   pwMinor    - [OUT] A pointer to the minor version number
+//   pwBuild    - [OUT] A pointer to the build number
+//   pwRevision - [OUT] A pointer to the revision number
+//   
+// Returns TRUE if successful
+//
+BOOL GetProductVersion(WORD * pwMajor, WORD * pwMinor, WORD * pwBuild, WORD * pwRevision)
+{
+    BOOL               success = FALSE;
+    WCHAR              wszCurrentModulePath[MAX_PATH];
+    DWORD              cchPath;
+    DWORD              cbVerInfo;
+    LPVOID             pFileVerInfo;
+    UINT               uLen;
+    VS_FIXEDFILEINFO * pFixedFileInfo;
+
+    cchPath = GetModuleFileName(NULL, wszCurrentModulePath, MAX_PATH);
+
+    if (cchPath && GetLastError() != ERROR_INSUFFICIENT_BUFFER)
+    {
+        cbVerInfo = GetFileVersionInfoSize(wszCurrentModulePath, NULL);
+
+        if (cbVerInfo)
+        {
+            pFileVerInfo = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, cbVerInfo);
+
+            if (pFileVerInfo)
+            {
+                if (GetFileVersionInfo(wszCurrentModulePath, 0, cbVerInfo, pFileVerInfo))
+                {
+                    // Get the pointer to the VS_FIXEDFILEINFO structure
+                    if (VerQueryValue(pFileVerInfo, L"\\", (LPVOID *)&pFixedFileInfo, &uLen))

```

Comment:
Since you are using VerQueryValue (not VerQueryValueW), is there a reason why you use `L"\\"`, rather than `TEXT("\\")`?

Reply:
Good point. I was assuming UNICODE would be set at build time, but I agree that using TEXT would be more consistent. I'll make that change and also use TCHAR rather than WCHAR.